shapes/neurosciencegraph/commons/list/schema.json

```diff
@@ -0,0 +1,359 @@
+{
+  "@context": [
+    "https://incf.github.io/neuroshapes/contexts/schema.json",
+    {
+      "this": "https://neuroshapes.org/commons/list/shapes/"
+    }
+  ],
+  "@id": "nsg:commons/list",
+  "@type": "nxv:Schema",
+  "shapes": [
+    {
+      "@id": "this:EmptyNode",
+      "hasValue": [
+        {
+          "@list": []
+        }
+      ],
+      "property": [
+        {
+          "maxCount": 0,
+          "path": "rdf:rest"
+        },
+        {
+          "maxCount": 0,
+          "path": "rdf:first"
+        }
+      ]
+    },
+    {
+      "@id": "this:DataNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        },
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:first"
+        }
+      ]
+    },
+    {
+      "@id": "this:IntegerNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        },
+        {
+          "datatype": "xsd:integer",
+          "maxCount": 1,
+          "message": "Data should be integer type",
+          "minCount": 1,
+          "path": "rdf:first"
+        }
+      ]
+    },
+    {
+      "@id": "this:StringNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "datatype": "xsd:string",
+          "maxCount": 1,
+          "message": "Data should be string type",
+          "minCount": 1,
+          "path": "rdf:first"
+        },
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        }
+      ]
+    },
+    {
+      "@id": "this:DoubleNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        },
+        {
+          "datatype": "xsd:double",
+          "maxCount": 1,
+          "message": "Data should be double type",
+          "minCount": 1,
+          "path": "rdf:first"
+        }
+      ]
+    },
+    {
+      "@id": "shsh:ListNodeShape",
```

Comment:
This ListNodeShape is not exactly the same as the one in https://www.w3.org/ns/shacl-shacl#. I would recommend to use this:ListNodeShape.